### PR TITLE
[ONNX] Softmax operator

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/softmax.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph/op/softmax.hpp"
+
+#include "ngraph/frontend/onnx_import/exceptions.hpp"
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/softmax.cpp]: #include "exceptions.hpp"

File: src/ngraph/frontend/onnx_import/op/softmax.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#include "ngraph/frontend/onnx_import/op/softmax.hpp"
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/softmax.cpp]: #include "op/softmax.hpp"

File: src/ngraph/frontend/onnx_import/op/softmax.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                    axis = data_shape.size() + axis;
+                }
+                else if (axis >= data_shape.size())
+                {
+                    throw error::parameter::Value(
+                        "Softmax node (",
+                        node.get_name(),
+                        "): provided axis attribute is out of input tensor dimensions range.");
+                }
+                std::vector<size_t> axes(data_shape.size() - axis);
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/softmax.cpp]: `std::vector<> axes{...};`

File: src/ngraph/frontend/onnx_import/op/softmax.cpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                else if (axis >= data_shape.size())
+                {
+                    throw error::parameter::Value(
+                        "Softmax node (",
+                        node.get_name(),
+                        "): provided axis attribute is out of input tensor dimensions range.");
+                }
+                // create vector of capacity data_dimensions - axis_divider position
+                std::vector<size_t> axes(data_shape.size() - axis);
+                std::iota(std::begin(axes), std::end(axes), axis);
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/softmax.cpp]: You need to add `<numeric>` when using `std::iota`

File: src/ngraph/frontend/onnx_import/op/softmax.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright 2018 Intel Corporation
```
Comment (rkimballn1) [inline src/ngraph/frontend/onnx_import/op/softmax.cpp]: Incorrect license header. Asterisks must all be in column zero

File: src/ngraph/frontend/onnx_import/op/softmax.hpp [hunk 6]
```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright 2018 Intel Corporation
```
Comment (rkimballn1) [inline src/ngraph/frontend/onnx_import/op/softmax.hpp]: This one too

**Pull request merged**